incuna_auth/middleware/login_required_feincms.py

```diff
@@ -0,0 +1,48 @@
+from django.conf import settings
+from django.core.exceptions import ImproperlyConfigured
+
+from .permission import FeinCMSPermissionMiddleware, LoginPermissionMiddlewareMixin
+
+
+def check_feincms_page():
+    """
+    Check that FeinCMSLoginRequiredMiddleware isn't being used without its dependency,
+    feincms.context_processors.add_page_if_missing.
+    """
+    processors = settings.TEMPLATE_CONTEXT_PROCESSORS
+
+    if 'feincms.context_processors.add_page_if_missing' in processors:
+        return
+
+    raise ImproperlyConfigured(
+        (
+            "TEMPLATE_CONTEXT_PROCESSORS does not contain add_page_if_missing."
+            "FeinCMSLoginRequiredMiddleware requires the FeinCMS page middleware"
+            "to be installed. Ensure your TEMPLATE_CONTEXT_PROCESSORS setting"
+            "includes 'feincms.context_processors.add_page_if_missing'."
+        ),
+    )
```

Comment:
You've also got implicit concatenation errors here:
`add_page_if_missing.FeincmsLoginRequiredMiddleware`
`middlewareto`
`settingincludes`